react/src/components/ImageList.tsx

```diff
@@ -0,0 +1,369 @@
+          a?.tag && b?.labels
+            ? getConstraints(
+                a?.tag,
+                a?.labels as { key: string; value: string }[],
+              )[0] || ''
+            : '';
+        const requirementB =
+          b?.tag && b?.labels
+            ? getConstraints(
+                b?.tag,
+                b?.labels as { key: string; value: string }[],
+              )[0] || ''
+            : '';
+        if (requirementA === '' && requirementB === '') return 0;
+        if (requirementA === '') return -1;
+        if (requirementB === '') return 1;
+        return requirementA.localeCompare(requirementB);
+      },
+      render: (text, row) =>
+        row?.tag ? (
+          <ConstraintTags
+            tag={row?.tag}
+            labels={row?.labels as { key: string; value: string }[]}
+          />
+        ) : null,
+    },
+    {
+      title: t('environment.Digest'),
+      dataIndex: 'digest',
+      key: 'digest',
+      sorter: (a, b) =>
+        a?.digest && b?.digest ? a.digest.localeCompare(b.digest) : 0,
+    },
+    {
+      title: t('environment.ResourceLimit'),
+      dataIndex: 'resource_limits',
+      key: 'resource_limits',
+      render: (text, row) =>
+        row?.resource_limits?.map((resource_limit) => (
+          <ResourceNumber
+            // @ts-ignore
+            type={resource_limit.key}
+            value={resource_limit?.min || '0'}
+            max={resource_limit?.max || ''}
+          />
+        )),
+    },
+    {
+      title: t('general.Control'),
+      key: 'control',
+      dataIndex: 'control',
+      fixed: 'right',
+      render: (text, row) => {
+        return (
+          <>
+            <Flex direction="row" align="stretch" justify="center" gap="xxs">
+              <Button
+                type="text"
+                icon={
+                  <SettingOutlined
+                    style={{
+                      color: token.colorInfo,
+                    }}
+                  />
+                }
+                onClick={() => setManagingResourceLimit(row)}
+              />
+              <Button
+                type="text"
+                icon={
+                  <AppstoreOutlined
+                    style={{
+                      color: token.colorInfo,
+                    }}
+                  />
+                }
+                onClick={() => {
+                  setManagingApp(row);
+                }}
+              />
+            </Flex>
+          </>
+        );
+      },
+    },
+  ];
+  return (
+    <>
+      <Flex
+        direction="column"
+        align="stretch"
+        style={{
+          flex: 1,
+          ...style,
+        }}
+      >
+        <Flex justify="end" style={{ padding: token.paddingSM }}>
+          <Button
+            icon={<VerticalAlignBottomOutlined />}
+            style={{ backgroundColor: token.colorPrimary, color: 'white' }}
+            onClick={() => {
+              setIsOpenInstallModal(true);
+            }}
```

Comment:
[related comment](https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2615/feat-migrate-image-list#comment-PRRC_kwDOCRTcws5mG_cB)


```suggestion
            onClick={() => {
              _.isEmpty(selectedRowKeys)
                ? message.warning(t('environment.NoImagesAreSelected'))
                : setIsOpenInstallModal(true);
            }}
```